contrib/pkg/utils/aws/aws.go

```diff
@@ -1,16 +1,214 @@
 package aws
 
 import (
+	"errors"
+	"fmt"
 	"os"
 	"path/filepath"
+	"strings"
 
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+
+	hivev1 "github.com/openshift/hive/apis/hive/v1"
 	"github.com/openshift/hive/contrib/pkg/utils"
 	"github.com/openshift/hive/pkg/constants"
 	log "github.com/sirupsen/logrus"
 	ini "gopkg.in/ini.v1"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+const (
+	// PrivateLinkHubAcctCredLabel is added to the AWS PrivateLink Hub account credentials Secret
+	// which is created by the "hiveutil awsprivatelink enable ..." command
+	// and referred to by HiveConfig.spec.awsPrivateLink.credentialsSecretRef.
+	PrivateLinkHubAcctCredLabel = "hive.openshift.io/awsprivatelink-hub-acct-credentials"
+)
+
+// AssociatedVpcWithClient encapsulates an associated VPC for PrivateLink and an AWS EC2 client for it
+type AssociatedVpcWithClient struct {
+	AWSEC2Client  *ec2.EC2
+	AssociatedVpc *hivev1.AWSPrivateLinkVPC
+}
+
+// GetAWSEC2Client gets an AWS EC2 client in a certain region.
+func GetAWSEC2Client(region string) *ec2.EC2 {
```

Comment:
I would prefer you reuse the existing [AWS client wrapper](https://github.com/openshift/hive/blob/4dfe9ecb15e1db492177e12ab1d5385b98092850/pkg/awsclient/client.go#L60). Add more wrapper methods to it if you need.

Reply:
Okay. The code is now updated.